File: Task 25/ClearHire-Job Portal/Backend/app/crud/interviews.py
from sqlalchemy.orm import Session

from app.models.interview import Interview


def create_interview(
    db: Session,
    application_id: int,
    interview_type: str,
    scheduled_at,
    duration_minutes: int | None,
    meeting_link: str | None,
    notes: str | None,
):
    interview = Interview(
        application_id=application_id,
        interview_type=interview_type,
        scheduled_at=scheduled_at,
        duration_minutes=duration_minutes,
        meeting_link=meeting_link,
        notes=notes,
    )

    db.add(interview)
    db.commit()
    db.refresh(interview)

    return interview


def get_interview(
    db: Session,
    interview_id: int,
):
    return (
        db.query(Interview)
        .filter(Interview.id == interview_id)
        .first()
    )


def get_interviews(
    db: Session,
):
    return db.query(Interview).all()


def get_interviews_by_application(
    db: Session,
    application_id: int,
):
    return (
        db.query(Interview)
        .filter(
            Interview.application_id == application_id
        )
        .all()
    )


def update_interview(
    db: Session,
    interview: Interview,
    data: dict,
):
    for field, value in data.items():
        setattr(interview, field, value)

    db.commit()
    db.refresh(interview)

    return interview


def delete_interview(
    db: Session,
    interview: Interview,
):
    db.delete(interview)
    db.commit()

File: Task 25/ClearHire-Job Portal/Frontend/frontend/src/services/interviews.js
import api from "./api";

export const createInterview = async (interviewData) => {
  const response = await api.post("/interviews", interviewData);
  return response.data;
};

export const getInterview = async (interviewId) => {
  const response = await api.get(`/interviews/${interviewId}`);
  return response.data;
};

export const getApplicationInterviews = async (applicationId) => {
  const response = await api.get(
    `/interviews/application/${applicationId}`
  );
  return response.data;
};

export const updateInterview = async (interviewId, interviewData) => {
  const response = await api.put(
    `/interviews/${interviewId}`,
    interviewData
  );
  return response.data;
};

export const deleteInterview = async (interviewId) => {
  await api.delete(`/interviews/${interviewId}`);
};